### PR TITLE
Php 74 type support

### PR DESCRIPTION
**Pull request by brendt**
As much support as possible for PHP 7.4 typed properties.

### CODE REVIEW

File: src/FieldValidator.php [hunk 1]
```diff
@@ -6,53 +6,47 @@
...
-        $definition = $matches[1] ?? '';
-
-        $this->hasTypeDeclaration = $definition !== '';
-        $this->hasDefaultValue = $hasDefaultValue;
-        $this->isNullable = $this->resolveNullable($definition);
-        $this->isMixed = $this->resolveIsMixed($definition);
-        $this->isMixedArray = $this->resolveIsMixedArray($definition);
-        $this->allowedTypes = $this->resolveAllowedTypes($definition);
-        $this->allowedArrayTypes = $this->resolveAllowedArrayTypes($definition);
+        if ($property->getDocComment()) {
```
Comment (brendt) [inline src/FieldValidator.php:35]: I deliberately chose to prioritise doccomments over types, in order to still support stuff like this:

```php
    /** @var int[] */
    public array $numbers;
```

If PHP ever adds generics, we can switch the priority of field validators.

**Comment by brendt** [comment]
Released as 2.1.0: https://github.com/spatie/data-transfer-object/releases/tag/2.1.0

File: src/FieldValidator.php (hunk 1 above)
Comment (mfn) [reply src/FieldValidator.php:35]: Purrfect 🐱

**Comment by devfrey** [comment]
@brendt Actually... it wasn't released as part of 2.1.0. Unless I'm seeing something wrong, it's unreleased on `master`.

**Comment by brendt** [comment]
@devfrey you're absolutely right…

I tagged 2.2.0, it contains the right commits! https://github.com/spatie/data-transfer-object/releases/tag/2.2.0

**Comment by devfrey** [comment]
@brendt Awesome, thanks!

**Pull request merged**